sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -112,8 +105,9 @@ private[sql] object FileSourceStrategy extends Strategy with Logging {
 
       val readFile = files.fileFormat.buildReader(
         sqlContext = files.sqlContext,
+        physicalSchema = files.dataSchema,
```

Comment:
To keep consistent with the previous `buildIntenralScan`, I prefer to call them `dataSchema` and `requiredSchema`